### PR TITLE
Add data-dir support

### PR DESCRIPTION
**Pull request by jakefhyde**
https://github.com/rancher/rancher/issues/45038

### CODE REVIEW

**Review by HarrisonWAffel — APPROVED** [review]

File: package/run.ps1 [hunk 1]
```diff
@@ -48,12 +48,13 @@ function Get-StringHash {
...
-$SAI_FILE_DIR = "c:/var/lib/rancher/rke2/system-agent-installer"
+$RKE2_DATA_DIR = if ($env:RKE2_DATA_DIR) { $env:RKE2_DATA_DIR } else { "c:/var/lib/rancher/rke2" };
+$SAI_FILE_DIR = "$RKE2_DATA_DIR/system-agent-installer"
 $RESTART_STAMP_FILE = "$SAI_FILE_DIR/rke2_restart_stamp"
 $PRIOR_RESTART_STAMP = ""
 $RESTART = $false
 
-New-Directroy -Path "c:/var/lib/rancher/rke2"
+New-Directroy -Path "$RKE2_DATA_DIR"
 New-Directroy $SAI_FILE_DIR
```
Comment (brandond) [inline package/run.ps1:58]: I am guessing that this is currently doing nothing because of the typo. Should be `New-Directory` not New-Directroy`

**Review by brandond — CHANGES_REQUESTED** [review]
typo nit

File: package/run.ps1 (hunk 1 above)
Comment (HarrisonWAffel) [reply package/run.ps1:58]: No idea why but a `New-Directroy` is defined on line 22, which defines the `-Path` parameter. I'm guessing it's to better handle the case where a directory already exists? 

https://github.com/rancher/system-agent-installer-rke2/blob/8d902f75a75ebb3d9059af4d02923dbad44927e6/package/run.ps1#L22-L32

Comment (brandond) [reply package/run.ps1]: Oh, more fun stuff left over from Jamie and the rush-job that was initial Windows support 😬 . Can we fix the typo and inconsistent use of -Path while we're here?

**Review by brandond — APPROVED** [review]

**Comment by jakefhyde** [comment]
@HarrisonWAffel re-requested your review just to make sure the `New-Directory` changes look sane.

**Review by HarrisonWAffel — APPROVED** [review]
lgtm

**Review by Oats87 — APPROVED** [review]

**Pull request merged**